packages/xml-views-validation/src/validators/attributes/unknown-attribute-key.ts

```diff
@@ -0,0 +1,179 @@
+import { includes, find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import {
+  UI5SemanticModel,
+  UI5Aggregation,
+  UI5Association,
+  UI5Event,
+  UI5Prop,
+} from "@ui5-language-assistant/semantic-model-types";
+import {
+  getUI5ClassByXMLElement,
+  getUI5AggregationByXMLElement,
+  isXMLNamespaceKey,
+  flattenProperties,
+  flattenEvents,
+  flattenAssociations,
+  flattenAggregations,
+  ui5NodeToFQN,
+} from "@ui5-language-assistant/logic-utils";
+import { UnknownAttributeKeyIssue } from "../../../api";
+import { TEMPLATING_NS, CUSTOM_DATA_NS } from "../../utils/special-namespaces";
+
+export function validateUnknownAttributeKey(
+  attribute: XMLAttribute,
+  model: UI5SemanticModel
+): UnknownAttributeKeyIssue[] {
+  if (attribute.syntax.key === undefined || attribute.key === null) {
+    // Can't give an error without a position or value
+    return [];
+  }
+
+  if (isAttributeNameAlwaysValid(attribute)) {
+    return [];
+  }
+
+  // Only validate (known) classes and aggregations
+  const xmlElement = attribute.parent;
+  const ui5Class = getUI5ClassByXMLElement(xmlElement, model);
+  let ui5Aggregation: UI5Aggregation | undefined = undefined;
+  if (ui5Class === undefined) {
+    ui5Aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  }
+
+  // Aggregations don't have additional allowed attributes
+  if (ui5Aggregation !== undefined) {
+    return [
+      {
+        kind: "UnknownAttributeKey",
+        message: `Unknown attribute key: ${attribute.key}`,
+        offsetRange: {
+          start: attribute.syntax.key.startOffset,
+          end: attribute.syntax.key.endOffset,
+        },
+        severity: "error",
+      },
+    ];
+  }
+
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  // Special case: "class" property can be used on all Controls and selected Elements.
```

Comment:
- This is a CSS class?
- Can we extract the white list to a record/array?

Reply:
e.g HTML props whitelist?

Reply:
Only `class` is supported, not all html properties.
HTML tags can be used but they are in a specific namespace (the details are in the task for validating tag names). We don't validate their attributes.
We can have a whitelist for the special properties, but there will still be the special case of custom data namespace (since it doesn't have an attribute name).

Reply:
And yes, it's a css class
https://openui5nightly.hana.ondemand.com/topic/b564935324f449209354c7e2f9903f22